### PR TITLE
feat: Bar Gauge Support scrolling

### LINKED ISSUE #17736: Bar gauge: Add scrolling support
Hi all. I know you are working on it and that this is a new visualisation type currently on beta. I thought I would share this with you sooner rather than later, hoping it is an issue easy enough to fix before the visualisation type goes GA.

Disclaimer: this was my very first time I used the bar gauge, if you think I am using the visualisation incorrectly (i.e. wrong query, wrong approach to visualising the data in question, etc), please let me know and advise. 

I was trying to visualise Kafka consumer lags for topics within Grafana using the bar gauge. I thought it would be nice to have multiple bars visually highlighting which topic has the most lag, given a time frame set.

After configuring the query, this is what I could see:
Screenshot 2019-06-24 at 16 44 23

Text as well as the bar and the number on the right become completely unreadable. 

The query I used is as follows:
```
SELECT mean("rep_lag") FROM "kafka_consumer_group" WHERE ("project" = '{{project}}' AND "service" = '{{service-name}}') AND $timeFilter GROUP BY time($__interval), "name" fill(null)
```

The data source used is InfluxDB and the data is coming out of a Kafka instance running on Aiven. Aiven themselves provide a nice line graph using the same query which produces the following visualisation:
Screenshot 2019-06-24 at 16 51 33

**Steps to reproduce**:
- Get a Kafka instance up and running on Aiven, backed by an InfluxDB instance containing Kafka metrics (also on Aiven)
- Create the InfluxDB data source within Grafana so that it can pull the data from the InfluxDB instance running in Aiven
- Create the bar gauge visualisation using the query above

**Actual Result**:
Please see screenshots above

**Expected Result**
Text, bars and numbers should be readable

**Relevant WCAG Criteria:** [1.4.4](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#resize-text)

**Environment**:
- Grafana version: 6.2.4
- Data source type & version: InfluxDB
- User OS & Browser: macOS, Chrome
- Others:

### PR DESCRIPTION
**Pull request by myml**
**What this PR does / why we need it**:
All of them are here https://github.com/grafana/grafana/issues/17736
>Currently it just divides the space by the number of results and gives each bar that much space. We could add a mode where you explicitly ask for XX pixels for each result, and get a scrollbar when the data overflows

>When you have a large set of data sources for (vertical) bar guages, some of guages fall hidden outside the panel and the only way to view them is to zoom-out the dashboard which is exteremely inconvenient.

Added 'min-width' and 'min-height' options. If the minimum value is set, scrolling on overflow panel.

image

image

**Which issue(s) this PR fixes**:

Fixes #17736

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/bargauge/module.tsx [hunk 1]
```diff
@@ -28,6 +28,20 @@ export const plugin = new PanelPlugin<BarGaugeOptions>(BarGaugePanel)
...
+        path: 'minVizWidth',
+        name: 'Min width',
+        description: 'minimum bar width',
+        defaultValue: 0,
+        showIf: (options: BarGaugeOptions) => options.orientation === VizOrientation.Vertical,
+      })
+      .addNumberInput({
+        path: 'minVizHeight',
+        name: 'Min height',
+        description: 'minimum bar height',
```
Comment (ashharrison90) [inline public/app/plugins/panel/bargauge/module.tsx:42]:
```suggestion
        description: 'Minimum row height',
```

**Review by ashharrison90 — COMMENTED** [review]
hi @myml 👋 

just a small tidy up to the description. i've suggested column/row instead of bar because in some cases the text could appear above the bar, so it's not strictly always correlating to the bar width/height.

otherwise i think this looks good. will discuss with the rest of the team to make sure there's no other concerns. 

thanks! 😄

**Comment by myml** [comment]
> hi @myml wave
> 
> just a small tidy up to the description. i've suggested column/row instead of bar because in some cases the text could appear above the bar, so it's not strictly always correlating to the bar width/height.
> 
> otherwise i think this looks good. will discuss with the rest of the team to make sure there's no other concerns.
> 
> thanks! smile

Perfection!

**Comment by ashharrison90** [comment]
@grafana/docs-squad are there any docs we should update for this change? 🤔

**Comment by achatterjee-grafana** [comment]
> @grafana/docs-squad are there any docs we should update for this change? 🤔

Hi @ashharrison90, since this looks like a new functionality, we will need to add/update documentation. Docs squad will be happy to collaborate. 

Hi @myml, will you be open to adding some documentation? We can help you with the style guide and information architecture issues. Let us know how we can help.

**Comment by myml** [comment]
Thank you @achatterjee-grafana. I have updated the documentation.
Can you help me to see what is wrong?

**Review by achatterjee-grafana — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
lgtm - thanks for a great contribution @myml 🙌

**Comment by torkelo** [comment]
@ashharrison90 did we really need two new UI options here? we could have just added some smarter handling/detection that bars are too small. Adding 2 new UI options for a sizing bug seems like a bad solution.

**Comment by torkelo** [comment]
@ashharrison90 no worries, but would have been good to start with better auto min constraints (and scrolling) than jumping directly to options.  Worked very hard to making gauge and bar gauge have very good auto size handling so there is less need for ui options

**Pull request merged**